libvast/test/dumper_plugin.cpp

```diff
@@ -0,0 +1,139 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <vast/collect.hpp>
+#include <vast/plugin.hpp>
+#include <vast/table_slice.hpp>
+#include <vast/test/test.hpp>
+
+#include <caf/test/dsl.hpp>
+
+#include <fcntl.h>
+#include <unistd.h>
+
+using namespace vast;
+
+namespace {
+
+struct fixture {
+  struct mock_control_plane : operator_control_plane {
+    [[nodiscard]] virtual auto self() noexcept -> caf::event_based_actor& {
+      FAIL("no mock implementation available");
+    }
+    virtual auto stop([[maybe_unused]] caf::error error = {}) noexcept -> void {
+      FAIL("no mock implementation available");
+    }
+
+    virtual auto warn([[maybe_unused]] caf::error warning) noexcept -> void {
+      FAIL("no mock implementation available");
+    }
+
+    virtual auto emit([[maybe_unused]] table_slice metrics) noexcept -> void {
+      FAIL("no mock implementation available");
+    }
+
+    [[nodiscard]] virtual auto
+    demand([[maybe_unused]] type schema = {}) const noexcept -> size_t {
+      FAIL("no mock implementation available");
+    }
+
+    [[nodiscard]] virtual auto schemas() const noexcept
+      -> const std::vector<type>& {
+      FAIL("no mock implementation available");
+    }
+
+    [[nodiscard]] virtual auto concepts() const noexcept
+      -> const concepts_map& {
+      FAIL("no mock implementation available");
+    }
+  };
+
+  fixture() {
+    // TODO: Move this into a separate fixture when we are starting to test more
+    // than one dumper type.
+    dumper_plugin = vast::plugins::find<vast::dumper_plugin>("stdout");
+    REQUIRE(dumper_plugin);
+    current_dumper = unbox(dumper_plugin->make_dumper({}, {}, control_plane));
+  }
+
+  ~fixture() {
+  }
```

Comment:
I think this can be removed

Reply:
True, actually. Removed the destructors in https://github.com/tenzir/vast/pull/3014/commits/42d15545115f201df6df061e86d30553fc2588b1